src/main/java/org/inferred/freebuilder/processor/DefaultPropertyFactory.java

```diff
@@ -169,18 +169,56 @@ public void addFinalFieldAssignment(SourceBuilder code, String finalField, Strin
 
     @Override
     public void addMergeFromValue(Block code, String value) {
-      code.addLine("%s(%s.%s());", setter(property), value, property.getGetterName());
+      Excerpt defaults = Declarations.freshBuilder(code, metadata).orNull();
```

Comment:
This is standard in this codebase, and follows the advice of http://blog.joda.org/2015/08/java-se-8-optional-pragmatic-approach.html. Were this Java 8 code, I would have used lambdas.